src/capsule/ChallengeStorage.ts

```diff
@@ -20,20 +24,60 @@ export interface Signature {
   recoveryParam: number
 }
 
-const ec = new elliptic.ec('p256')
+const ecl = new elliptic.ec('p256')
 
-const privateKey = '202d73cbde65f547c75613ace311393ac97f2556cbe3aca32bf48eb84ec2198c'
+const PEM_HEADER = '-----BEGIN PUBLIC KEY-----'
+const PEM_FOOTER = '-----END PUBLIC KEY-----'
 export class ChallengeReactNativeStorage extends ChallengeStorage {
+  private storageIdentifier() {
+    return 'challenge-' + this.userId
+  }
+  async getPublicKey(): Promise<string> {
+    const pemPublicKey = await DeviceCrypto.getOrCreateAsymmetricKey(this.storageIdentifier(), {
+      accessLevel: AccessLevel.ALWAYS,
+      invalidateOnNewBiometry: false,
+    })
+
+    const base64PublicKey = pemPublicKey.replace(PEM_FOOTER, '').replace(PEM_HEADER, '').trim()
+    const bufferPublicKey = Buffer.from(base64PublicKey, 'base64')
+    const publicKeyHexAsnPreamble = bufferPublicKey.toString('hex')
+    const publicKeyHex = publicKeyHexAsnPreamble.slice(52)
+    return publicKeyHex
+  }
+
+  async signChallenge(message: string): Promise<Signature> {
+    const signatureDERBase64 = await DeviceCrypto.sign(this.storageIdentifier(), message, {
+      biometryTitle: 'Authenticate',
+      biometrySubTitle: 'Signing',
+      biometryDescription: 'Authenticate your self to sign the text',
+    })
+
+    const signatureDERBuffer = Buffer.from(signatureDERBase64, 'base64')
+    const signatureDERHex = signatureDERBuffer.toString('hex')
+    const signature = new EllipticSignature(signatureDERHex, 'hex') as ec.Signature // hack due to incorrect typings
+    const cannonicalSignature = {
+      r: signature.r.toString('hex'),
+      s: signature.s.toString('hex'),
+      recoveryParam: signature.recoveryParam as number,
+    }
+    return cannonicalSignature
+  }
+}
+
+export class ChallengeFakeReactNativeStorage extends ChallengeStorage {
```

Comment:
Can you add a comment header to describe the purpose of this? Is this just to test locally without a real device?